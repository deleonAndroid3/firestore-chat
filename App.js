import 'react-native-gesture-handler'
import React, { useEffect, useState } from 'react'
import { CameraRoll, LogBox } from 'react-native'
import { firebase, firestore } from './src/firebase/firebase.app'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'
import { LoginScreen, ChatScreen, RegistrationScreen } from './src/screens'

import { decode, encode } from 'base-64'
if (!global.btoa) {
  global.btoa = encode
}
if (!global.atob) {
  global.atob = decode
}

LogBox.ignoreLogs([
  new RegExp('Setting a timer for a long period of time'),
  new RegExp('Remote debugger'),
])

const Stack = createStackNavigator()

export function test(x,y) {
  return x + y
}

export default function App() {
  const [loading, setLoading] = useState(false)
  const [user, setUser] = useState(null)

  useEffect(() => {
    firebase.auth().onAuthStateChanged((user) => {
      if (user) {
        firestore
          .collection('users')
          .doc(user.uid)
          .get()
          .then((document) => {
            const userData = document.data()
            setLoading(false)
            setUser(userData)
          })
          .catch((error) => {
            console.log('There was a problem on getting the user.', error)
            setLoading(false)
          })
      } else {
        setLoading(false)
      }
    })
  }, [])

  if (loading) {
    return <></>
  }

  return (
    <NavigationContainer>
      <Stack.Navigator>
        {user ? (
          <Stack.Screen name='Home'>
            {(props) => <ChatScreen {...props} userData={user} />}
          </Stack.Screen>
        ) : (
          <>
            <Stack.Screen name='Login' component={LoginScreen} />
            <Stack.Screen name='Registration' component={RegistrationScreen} />
          </>
        )}
      </Stack.Navigator>
    </NavigationContainer>
  )
}
