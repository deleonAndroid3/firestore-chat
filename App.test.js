/* eslint-disable no-undef */
import React from 'react'
import { create, act } from 'react-test-renderer'
import App from './App'
it('App renders without crashing', () => {
  let rendered
  act(() => {
    rendered = create(<App />)
  })
  expect(rendered.toJSON()).toBeTruthy()
})

it('App test against snapshot', () => {
  let tree
  act(() => {
    tree = create(<App />)
  })
  expect(tree.toJSON()).toMatchSnapshot()
})

