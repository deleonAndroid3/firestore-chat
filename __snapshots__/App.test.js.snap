// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test against snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "zIndex": 1,
          },
          false,
        ]
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                Login
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "flex": 1,
                        }
                      }
                    >
                      <RCTScrollView
                        automaticallyAdjustContentInsets={false}
                        contentInset={
                          Object {
                            "bottom": 0,
                          }
                        }
                        enableAutomaticScroll={true}
                        enableOnAndroid={false}
                        enableResetScrollToCoords={true}
                        extraHeight={75}
                        extraScrollHeight={0}
                        getScrollResponder={[Function]}
                        handleOnScroll={[Function]}
                        keyboardDismissMode="interactive"
                        keyboardOpeningTime={250}
                        keyboardShouldPersistTaps="always"
                        keyboardSpace={0}
                        onScroll={[Function]}
                        resetKeyboardSpace={[Function]}
                        scrollEventThrottle={1}
                        scrollForExtraHeightOnAndroid={[Function]}
                        scrollIntoView={[Function]}
                        scrollToEnd={[Function]}
                        scrollToFocusedInput={[Function]}
                        scrollToPosition={[Function]}
                        showsVerticalScrollIndicator={true}
                        style={
                          Object {
                            "flex": 1,
                            "width": "100%",
                          }
                        }
                        update={[Function]}
                        viewIsInsideTabBar={false}
                      >
                        <View>
                          <Image
                            source={1}
                            style={
                              Object {
                                "alignSelf": "center",
                                "flex": 1,
                                "height": 120,
                                "margin": 30,
                                "width": 90,
                              }
                            }
                          />
                          <TextInput
                            allowFontScaling={true}
                            autoCapitalize="none"
                            onChangeText={[Function]}
                            placeholder="E-mail"
                            placeholderTextColor="#aaaaaa"
                            rejectResponderTermination={true}
                            style={
                              Object {
                                "backgroundColor": "white",
                                "borderRadius": 5,
                                "height": 48,
                                "marginBottom": 10,
                                "marginLeft": 30,
                                "marginRight": 30,
                                "marginTop": 10,
                                "overflow": "hidden",
                                "paddingLeft": 16,
                              }
                            }
                            underlineColorAndroid="transparent"
                            value=""
                          />
                          <TextInput
                            allowFontScaling={true}
                            autoCapitalize="none"
                            onChangeText={[Function]}
                            placeholder="Password"
                            placeholderTextColor="#aaaaaa"
                            rejectResponderTermination={true}
                            secureTextEntry={true}
                            style={
                              Object {
                                "backgroundColor": "white",
                                "borderRadius": 5,
                                "height": 48,
                                "marginBottom": 10,
                                "marginLeft": 30,
                                "marginRight": 30,
                                "marginTop": 10,
                                "overflow": "hidden",
                                "paddingLeft": 16,
                              }
                            }
                            underlineColorAndroid="transparent"
                            value=""
                          />
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#788eec",
                                "borderRadius": 5,
                                "height": 48,
                                "justifyContent": "center",
                                "marginLeft": 30,
                                "marginRight": 30,
                                "marginTop": 20,
                                "opacity": 1,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "white",
                                  "fontSize": 16,
                                  "fontWeight": "bold",
                                }
                              }
                            >
                              Log in
                            </Text>
                          </View>
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "flex": 1,
                                "marginTop": 20,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#2e2e2d",
                                  "fontSize": 16,
                                }
                              }
                            >
                              Don't have an account?
                               
                              <Text
                                onPress={[Function]}
                                style={
                                  Object {
                                    "color": "#788eec",
                                    "fontSize": 16,
                                    "fontWeight": "bold",
                                  }
                                }
                              >
                                Sign up
                              </Text>
                            </Text>
                          </View>
                        </View>
                      </RCTScrollView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
